ttnn/cpp/ttnn/operations/pool/maxpool/device/max_pool2d_device_op.cpp

```diff
@@ -26,10 +26,12 @@ void validate_maxpool(const Tensor& input, const sliding_window::SlidingWindowCo
     TT_FATAL(is_pow2, "Row size (nchannels * bytes = {}) should be power of 2 ({}).", in_nbytes_c, is_pow2);
 
     TT_FATAL(input.memory_config().is_sharded(), "Input needs to be sharded");
-    TT_FATAL(input.memory_config().memory_layout == TensorMemoryLayout::HEIGHT_SHARDED, "Only height sharded tensors are supported.");
+    TT_FATAL((input.memory_config().memory_layout == TensorMemoryLayout::HEIGHT_SHARDED) ||
```

Comment:
if there is more limits to width sharding like special tensor dims or other stuff end it here with TT_FATAL